Modules/QuestieValidateGameCache.lua

```diff
@@ -0,0 +1,145 @@
+--[[
+Flow:
+-> QuestieValidateGameCache.StartCheck()
+--> Wait for PLAYER_ENTERING_WORLD
+---> Wait For QUEST_LOG_UPDATE (2x at login, 1x at reload)
+----> Game cache should have all quests in it, data of each quest may be invalid.
+      If data is invalid, Wait for next QUEST_LOG_UPDATE and check again.
+-----> Game Cache ok. Call possible callback functions.
+]]--
+
+---@class QuestieValidateGameCache
+local QuestieValidateGameCache = QuestieLoader:CreateModule("QuestieValidateGameCache")
+
+local stringSub = string.sub
+local GetNumQuestLogEntries, GetQuestLogTitle, GetQuestObjectives = GetNumQuestLogEntries, GetQuestLogTitle, C_QuestLog.GetQuestObjectives
+
+local tpack = table.pack or function(...) return { n = select("#", ...), ... } end
+local tunpack = table.unpack or unpack
+
+-- 3 * (Max possible number of quests in game quest log)
+-- This is a safe value, even smaller would be enough. Too large won't effect performance
+local MAX_QUEST_LOG_INDEX = 75
+
+local eventFrame
+local numberOfQuestLogUpdatesToSkip
+
+local cacheGood = false
```

Comment:
```suggestion
local isCacheGood = false
```

Reply:
I am still thinking is starting variable names with `is` a good thing or should that be just for functions/"methods" ?

Reply:
Done in 78b16dab24437ff7c33ec1111cd51a01e38c7ea2